libs/features/personalization/personalization.js

```diff
@@ -99,18 +99,105 @@ export const preloadManifests = ({ targetManifests = [], persManifests = [] }) =
 
 export const getFileName = (path) => path?.split('/').pop();
 
-const createFrag = (el, url, manifestId) => {
-  let href = url;
+export async function initDelayedModal({
+  delay,
+  displayMode,
+  hash,
+  contentUrl,
+  DELAYED_MODAL_DISPLAY_MODE,
+}) {
+  if (!delay || !displayMode || !hash || !contentUrl || !DELAYED_MODAL_DISPLAY_MODE) return;
+  const { findDetails, getModal } = await import('../../blocks/modal/modal.js');
+  const details = findDetails(hash);
+  if (!details?.id || !details?.path) return;
+  const delayInMiliseconds = delay * 1000;
+  const { miloLibs } = getConfig();
+  loadLink(`${miloLibs}/blocks/modal/modal.css`, { rel: 'stylesheet' });
+  if (displayMode === DELAYED_MODAL_DISPLAY_MODE.oncePerPageLoad) {
+    setTimeout(() => {
+      getModal(details);
+    }, delayInMiliseconds);
+  } else if (displayMode === DELAYED_MODAL_DISPLAY_MODE.oncePerSession) {
+    if (!window.sessionStorage.getItem('wasDelayedModalShown')) {
+      setTimeout(() => {
+        getModal(details);
+        window.sessionStorage.setItem('wasDelayedModalShown', true);
```

Comment:
this implementation supports one single delayed modal.
if the user navigates from one page to another in the same session, I think the other modals will not be shown.

Reply:
@yesil 
I created two test pages to verify this:
1. [First page](https://mwpw-140481-delayed-modal--milo--mirafedas.hlx.live/drafts/mirafedas/showdelayedmodalhere?manifest=%2Ffragments%2Fmirafedas%2Fpromos%2F2024%2Fglobal%2Fdelayed-modal-promo%2Fdelayed-modal-promo.json--all)
2. [Second page](https://mwpw-140481-delayed-modal--milo--mirafedas.hlx.page/drafts/mirafedas/secondpage)
On both pages I placed some traditional modals.
To me all works as expected.
If you want to test yourself, feel free to change the `display` param to either `pageload` or `session` in [this file](https://adobe.sharepoint.com/:x:/r/sites/adobecom/_layouts/15/Doc.aspx?sourcedoc=%7B58183116-E8D6-42D1-A007-82226E8405CA%7D&file=delayed-modal-promo.xlsx&action=default&mobileredirect=true)

Reply:
this test doesn't cover oncePerSession option.
please use oncePerSession, and link one page to the other one and try to navigate.
I expect the second modal will not be shown.